core/context/context.go

```diff
@@ -2,12 +2,18 @@ package context
 
 type contextKey uint8
 
-const (
-	ExecModeKey  contextKey = iota
-	CometInfoKey contextKey = iota
+type (
+	execModeKey    struct{}
+	cometInfoKey   struct{}
```

Comment:
I still don't know if it's right for comet to leak here; I'll approve but could we open an issue so we can discuss this in depth?